tools/depends/native/TexturePacker/src/decoder/KTXDecoder.cpp

```diff
@@ -0,0 +1,269 @@
+  {0x93D1, {KD_TEX_FMT_ASTC_LDR_5x4}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_5x4_KHR
+  {0x93D2, {KD_TEX_FMT_ASTC_LDR_5x5}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_5x5_KHR
+  {0x93D3, {KD_TEX_FMT_ASTC_LDR_6x5}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_6x5_KHR
+  {0x93D4, {KD_TEX_FMT_ASTC_LDR_6x6}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_6x6_KHR
+  {0x93D5, {KD_TEX_FMT_ASTC_LDR_8x5}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_8x5_KHR
+  {0x93D6, {KD_TEX_FMT_ASTC_LDR_8x6}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_8x6_KHR
+  {0x93D7, {KD_TEX_FMT_ASTC_LDR_8x8}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_8x8_KHR
+  {0x93D8, {KD_TEX_FMT_ASTC_LDR_10x5}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_10x5_KHR
+  {0x93D9, {KD_TEX_FMT_ASTC_LDR_10x6}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_10x6_KHR
+  {0x93DA, {KD_TEX_FMT_ASTC_LDR_10x8}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_10x8_KHR
+  {0x93DB, {KD_TEX_FMT_ASTC_LDR_10x10}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_10x10_KHR
+  {0x93DC, {KD_TEX_FMT_ASTC_LDR_12x10}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_12x10_KHR
+  {0x93DD, {KD_TEX_FMT_ASTC_LDR_12x12}}, // GL_COMPRESSED_SRGB8_ALPHA8_ASTC_12x12_KHR,
+};
+
+static const std::map<std::string, KD_TEX_SWIZ> SwizzleKeys
+{
+  {"KD_TEX_SWIZ_RGBA", KD_TEX_SWIZ_RGBA}, 
+  {"KD_TEX_SWIZ_RGB1", KD_TEX_SWIZ_RGB1},
+  {"KD_TEX_SWIZ_RRR1", KD_TEX_SWIZ_RRR1}, 
+  {"KD_TEX_SWIZ_111R", KD_TEX_SWIZ_111R},
+  {"KD_TEX_SWIZ_RRRG", KD_TEX_SWIZ_RRRG}, 
+  {"KD_TEX_SWIZ_RRRR", KD_TEX_SWIZ_RRRR},
+  {"KD_TEX_SWIZ_GGG1", KD_TEX_SWIZ_GGG1}, 
+  {"KD_TEX_SWIZ_111G", KD_TEX_SWIZ_111G},
+  {"KD_TEX_SWIZ_GGGA", KD_TEX_SWIZ_GGGA}, 
+  {"KD_TEX_SWIZ_GGGG", KD_TEX_SWIZ_GGGG},
+  {"KD_TEX_SWIZ_SDF", KD_TEX_SWIZ_SDF},   
+  {"KD_TEX_SWIZ_RGB_SDF", KD_TEX_SWIZ_RGB_SDF},
+  {"KD_TEX_SWIZ_MSDF", KD_TEX_SWIZ_MSDF},
+};
+
+static const std::map<std::string, KD_TEX_ALPHA> AlphaKeys
+{
+  {"KD_TEX_ALPHA_STRAIGHT", KD_TEX_ALPHA_STRAIGHT},
+  {"KD_TEX_ALPHA_OPAQUE", KD_TEX_ALPHA_OPAQUE},
+  {"KD_TEX_ALPHA_PREMULTIPLIED", KD_TEX_ALPHA_PREMULTIPLIED},
+};
+// clang-format on
+} // namespace
+
+KTXDecoder::KTXDecoder()
+{
+  m_extensions.emplace_back(".ktx");
+  m_substitutions.emplace_back(".astc.ktx", KD_TEX_FMT_ASTC_LDR);
+  m_substitutions.emplace_back(".bptc.ktx", KD_TEX_FMT_BPTC);
+  m_substitutions.emplace_back(".etc1.ktx", KD_TEX_FMT_ETC1);
+  m_substitutions.emplace_back(".etc2.ktx", KD_TEX_FMT_ETC2);
+  m_substitutions.emplace_back(".rgtc.ktx", KD_TEX_FMT_RGTC);
+  m_substitutions.emplace_back(".s3tc.ktx", KD_TEX_FMT_S3TC);
+}
+
+bool KTXDecoder::CanDecode(const std::string& filename)
+{
+  CFile fp;
+  if (!fp.Open(filename))
+    return false;
+
+  std::array<unsigned char, 12> magic;
+  if (fp.Read(&magic, sizeof(magic)) != sizeof(magic))
+    return false;
+
+  if (magic != m_magic)
+    return false;
+
+  std::array<unsigned char, 4> endianess;
+  if (fp.Read(&endianess, sizeof(endianess)) != sizeof(endianess))
+    return false;
+
+  if (endianess != m_endianess)
+    return false;
+
+  return true;
+}
```

Comment:
Sure, but does it matter for such an auxiliary tool?

Reply:
In terms of performance? No. Is it more readable? I'd say yes, but of course that's a matter of taste. I'm fine either way and won't oppose this PR because of this.